### PR TITLE
[JENKINS-39743] Create test for the javadoc retention for each build

### PR DESCRIPTION
**Pull request by varyvol**
[JENKINS-39743](https://issues.jenkins-ci.org/browse/JENKINS-39743)

@reviewbybees

### CODE REVIEW

File: src/test/java/plugins/JavadocPluginTest.java [hunk 1]
```diff
@@ -35,11 +39,30 @@ public void publish_javadoc_from_matrix_job() {
...
+    public void validate_javadoc_retention() {
+        FreeStyleJob job = jenkins.jobs.create();
+        setup(job);
+        Build b = job.startBuild().shouldSucceed();
+
+        assertThat(job, hasAction(JAVADOC_ACTION));
+        try {
+            assertThat(b, hasAction(JAVADOC_ACTION));
+            Assert.fail("Build #1 should not have Javadoc action");
+        } catch (AssertionError ex) {}
```
Comment (mafraba) [inline src/test/java/plugins/JavadocPluginTest.java]: Suggestion. Could this whole try block simply be replaced for the following? 

`assertThat("Build #1 should not have Javadoc action", b, not(hasAction(JAVADOC_ACTION)))`

Comment (olivergondza) [inline src/test/java/plugins/JavadocPluginTest.java]: It should, that is the reason we prefer hamcrest.

Comment (varyvol) [inline src/test/java/plugins/JavadocPluginTest.java]: I did not know that not. Thanks for the suggestion, changing it.

**Comment by olivergondza** [comment]
:+1:

**Comment by mafraba** [comment]
🐝

**Pull request merged**